### PR TITLE
Introduce `log_post_training_metrics` to enable/disable evaluation metrics autologging

### PR DESCRIPTION
**Pull request by harupy**
Signed-off-by: harupy <hkawamura0130@gmail.com>

## What changes are proposed in this pull request?

Introduce `log_eval_metrics` to enable/disable evaluation metrics autologging.

## How is this patch tested?

Unit tests

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/pyspark/ml/__init__.py [hunk 1]
```diff
@@ -666,6 +666,7 @@ def autolog(
     exclusive=False,
     disable_for_unsupported_versions=False,
     silent=False,
+    log_eval_metrics=True,
```
Comment (harupy) [inline mlflow/pyspark/ml/__init__.py:669]:
```suggestion
    log_post_training_metrics=True,
```

Should this param be renamed to `log_post_training_metrics`?

Comment (dbczumar) [reply mlflow/pyspark/ml/__init__.py:669]: Yep! Sounds good to me!

File: mlflow/pyspark/ml/__init__.py [hunk 2]
```diff
@@ -775,6 +776,8 @@ def autolog(
     :param silent: If ``True``, suppress all event logs and warnings from MLflow during pyspark ML
                    autologging. If ``False``, show all events and warnings during pyspark ML
                    autologging.
+    :param log_eval_metrics: If ``True``, evaluation metrics are logged. Defaults to ``True``.
+                             See the post training metrics section for details.
```
Comment (dbczumar) [inline mlflow/pyspark/ml/__init__.py:780]: Can we use the term `post training metrics` rather than `evaluation` metrics and clarify that these are logged when users call metric APIs such as `Evaluator.evaluate()`?

File: tests/sklearn/test_sklearn_autolog.py [hunk 3]
```diff
@@ -1859,3 +1859,23 @@ def test_is_metrics_value_loggable():
     assert not is_metric_value_loggable(np.bool(True))
     assert not is_metric_value_loggable([1, 2])
     assert not is_metric_value_loggable(np.array([1, 2]))
+
+
+@pytest.mark.parametrize("log_eval_metrics", [True, False])
+def test_log_eval_metrics_configuration(log_eval_metrics):
```
Comment (dbczumar) [inline tests/sklearn/test_sklearn_autolog.py:1865]: Can we verify that `log_eval_metrics` can be toggled on / off and add similar verification for `pyspark.ml`?

File: mlflow/pyspark/ml/__init__.py (hunk 2 above)
Comment (harupy) [reply mlflow/pyspark/ml/__init__.py:780]:
```suggestion
    :param log_post_training_metrics: If ``True``, post training metrics are logged. Defaults to
                                      ``True``. See the `post training metrics`_ for more details.

# `post training metrics`_ is rendered as a link
```

Can we just insert a link to the post training section?

File: tests/sklearn/test_sklearn_autolog.py (hunk 3 above)
Comment (harupy) [reply tests/sklearn/test_sklearn_autolog.py:1865]: @dbczumar You mean a test like https://github.com/mlflow/mlflow/pull/4705/files#diff-bd11e9a3dea4e2026366976e469085d6354fe088f0a933bc43a56177bc863376R763-R780?

File: mlflow/pyspark/ml/__init__.py (hunk 2 above)
Comment (dbczumar) [reply mlflow/pyspark/ml/__init__.py]: sure!

Comment (harupy) [reply mlflow/pyspark/ml/__init__.py]: @dbczumar Could you take a look at this page? https://10907-136202695-gh.circle-artifacts.com/0/docs/build/html/python_api/mlflow.sklearn.html#post-training-metrics

**Review by dbczumar — APPROVED** [review]
LGTM! Thanks @harupy !
